pandas/tests/indexing/multiindex/test_getitem.py

```diff
@@ -229,6 +229,30 @@ def test_frame_getitem_nan_multiindex(nulls_fixture):
     tm.assert_frame_equal(result, expected)
 
 
+def test_frame_getitem_nan_cols_multiindex(nulls_fixture):
```

Comment:
can you add all of the cases in the original post (working an non-working), there are 5 cases i think. pls parameterize

Reply:
Thank you for the suggestion. The additional cases were added and parameterized.

Keyword argument `check_column_type` was needed when slicing with `(["b"], [np.nan]),` in test case 5, the reason is explained below:

__Test Case 5__:
When asserting the types of the columns of the actual and expected result for the second level of the multiindex, on `_check_types` of `assert_index_equal` on `asserters.py` the left argument that is based on the sliced multiindex (argument `left` in `_check_types`) is:
- left: `Index(['bar', 'foo'], dtype='object')`

while the right argument that is based on the expected result is:
- right: `Index([], dtype='object')`

Based on that the left has "string" as inferred type, while the right one is empty which causes the test to fail, even though the resulting dataframes are identical.

I think this is because when slicing a dataframe with a multi-index the resulting levels of columns are the initial ones present in the original dataframe and are not updated.

In order to avoid that comparison I passed `check_column_type=False` as keyword argument.